modules/common/src/Storage/SelectFactory.php

```diff
@@ -226,51 +229,49 @@ private function propertyToString($property) {
   /**
    * Set filter conditions on DB query.
    *
-   * @param Drupal\Core\Database\Query\Select $db_query
-   *   A Drupal database query API object.
    * @param Drupal\common\Storage\Query $query
    *   A DKAN query object.
    */
-  private function setQueryConditions(Select $db_query, Query $query) {
+  private function setQueryConditions(Query $query) {
     foreach ($query->conditions as $c) {
       if (isset($c->groupOperator)) {
-        $this->addConditionGroup($db_query, $c);
+        $this->addConditionGroup($this->dbQuery, $c);
       }
       else {
-        $this->addCondition($db_query, $c);
+        $this->addCondition($this->dbQuery, $c);
       }
     }
   }
 
   /**
    * Add a condition to the DB query object.
    *
-   * @param mixed $db_query
+   * @param \Drupal\Core\Database\Query\Select|\Drupal\Core\Database\Query\Condition $statementObj
    *   Drupal DB API select object or condition object.
    * @param object $condition
    *   A condition from the DKAN query object.
    */
-  private function addCondition($db_query, $condition) {
+  private function addCondition($statementObj, $condition) {
     if (!isset($condition->operator)) {
       $condition->operator = '=';
     }
     $field = (isset($condition->collection) ? $condition->collection : $this->alias)
       . '.'
       . $condition->property;
-    $db_query->condition($field, $condition->value, strtoupper($condition->operator));
+    $statementObj->condition($field, $condition->value, strtoupper($condition->operator));
   }
 
   /**
    * Add a condition group to the database query.
    *
-   * @param Drupal\Core\Database\Query\Select|Drupal\Core\Database\Query\Condition $db_query
+   * @param Drupal\Core\Database\Query\Select|Drupal\Core\Database\Query\Condition $statementObj
    *   Drupal DB API select object.
    * @param object $conditionGroup
    *   A condition from the DKAN query object.
    */
-  private function addConditionGroup($db_query, $conditionGroup) {
+  private function addConditionGroup($statementObj, $conditionGroup) {
     $groupMethod = "{$conditionGroup->groupOperator}ConditionGroup";
-    $group = $db_query->$groupMethod();
+    $group = $this->dbQuery->$groupMethod();
```

Comment:
In case you were wondering, all this variable/property refactoring in this class is because condition objects are not supposed to be instantiated with the `new` keyword anymore, which is what happens when you run one of these condition group methods directly from a condition object. I think it's just a bug in core but whatever it's about this fixes it; always calling the method directly from the main Select object (these methods are used to both add conditions directly to the select object and to recursively add nested conditions to other conditions)